pootle/apps/pootle_store/forms.py

```diff
@@ -373,3 +407,129 @@ def save(self, **kwargs):
             super(UnitCommentForm, self).save(**kwargs)
 
     return UnitCommentForm
+
+
+class UnitSearchForm(forms.Form):
+
+    initial = forms.BooleanField(required=False)
+    count = forms.IntegerField(required=False)
+    path = forms.CharField(
+        max_length=2048,
+        required=True)
+    uids = MultipleArgsField(
+        field=forms.IntegerField(),
+        required=False)
+    filter = forms.ChoiceField(
+        required=False,
+        choices=UNIT_SEARCH_FILTER_CHOICES)
+    checks = forms.MultipleChoiceField(
+        required=False,
+        widget=CommaSeparatedCheckboxSelectMultiple,
+        choices=check_names.items())
+    category = CategoryChoiceField(
+        required=False,
+        choices=CATEGORY_CODES.items())
+    month = forms.DateField(
+        required=False,
+        input_formats=['%Y-%m'])
+    sort = forms.ChoiceField(
+        required=False,
+        choices=UNIT_SEARCH_SORT_CHOICES)
+
+    user = forms.ModelChoiceField(
+        queryset=get_user_model().objects.all(),
+        required=False,
+        to_field_name="username")
+
+    search = forms.CharField(
+        required=False,
+        widget=forms.TextInput(attrs={
+            'size': '15',
+            'placeholder': _('Search'),
+            'title': _("Search (Ctrl+Shift+S)<br/>Type and press Enter to "
+                       "search")}))
+
+    soptions = forms.MultipleChoiceField(
+        required=False,
+        widget=forms.CheckboxSelectMultiple,
+        choices=(
+            ('exact', _('Exact Match'))))
+
+    sfields = forms.MultipleChoiceField(
+        required=False,
+        widget=CommaSeparatedCheckboxSelectMultiple,
+        choices=(
+            ('source', _('Source Text')),
+            ('target', _('Target Text')),
+            ('notes', _('Comments')),
+            ('locations', _('Locations'))),
+        initial=['source', 'target'])
+
+    def __init__(self, *la, **kwa):
+        self.request_user = kwa.pop("user")
+        super(UnitSearchForm, self).__init__(*la, **kwa)
+        self.fields["modified-since"] = ISODateTimeField(required=False)
```

Comment:
Any reason to inject the field here? If yes please put some comment in code explaining why.